### PR TITLE
[MAIN] [STRATCONN] 3313 add page view field in setConfigurationFields

### PR DESCRIPTION
**Pull request by AnkitSegment**
Added New Send page view field in setConfigurationFields mapping, so that user can modify send_page_view when required. 

Jira ticket: 
https://segment.atlassian.net/browse/STRATCONN-3313

Test Screenshot

Screenshot 2024-03-01 at 11 22 45 AM

Screenshot 2024-03-01 at 11 18 11 AM

Screenshot 2024-03-01 at 11 17 56 AM

_A summary of your pull request, including the what change you're making and why._

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [X] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [X] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [X] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by Innovative-GauravKochar** [comment]
@AnkitSegment Nice Work , LGTM !

**Review by Innovative-GauravKochar — APPROVED** [review]

File: packages/browser-destinations/destinations/google-analytics-4-web/src/__tests__/setConfigurationFields.test.ts [hunk 1]
```diff
@@ -615,4 +637,52 @@ describe('Set Configuration Fields action', () => {
...
+    })
+
+    setConfigurationEvent.page?.(context)
+    expect(mockGtag).toHaveBeenCalledWith('config', 'G-XXXXXXXXXX', {
+      allow_ad_personalization_signals: false,
+      allow_google_signals: false,
+      send_page_view: true
+    })
+  })
+  it('should update config if payload has send_page_view true', () => {
```
Comment (varadarajan-tw) [inline packages/browser-destinations/destinations/google-analytics-4-web/src/__tests__/setConfigurationFields.test.ts:656]: Shouldn't this scenario not update config? I assume send_page_view is true by default.
```suggestion
  it('should not update config if payload has send_page_view true', () => {
```

**Review by varadarajan-tw — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
PR deployed earlier this week.

File: packages/browser-destinations/destinations/google-analytics-4-web/src/setConfigurationFields/index.ts [hunk 2]
```diff
@@ -170,6 +179,10 @@ const action: BrowserActionDefinition<Settings, Function, Payload> = {
     if (settings.pageView != true) {
       config.send_page_view = settings.pageView ?? true
     }
+
+    if (payload.send_page_view != true) {
+      config.send_page_view = payload.send_page_view ?? true
```
Comment (morrismatts) [inline packages/browser-destinations/destinations/google-analytics-4-web/src/setConfigurationFields/index.ts:184]: @AnkitSegment @joe-ayoub-segment Shouldn't this be falling back to the setting before defaulting to `true`?
```
config.send_page_view = payload.send_page_view ?? settings.pageView ?? true
```

In the case where `payload.send_page_view` is undefined and `settings.pageView` is false, the config should have `send_page_view` as false, right?

If I'm understanding correctly, this change broke our Segment <=> GA4 integration where we have `settings.pageView` set to false. It started sending page view events we were previously deliberately avoiding when this got published.

Comment (joe-ayoub-segment) [reply packages/browser-destinations/destinations/google-analytics-4-web/src/setConfigurationFields/index.ts:184]: Thanks @morrismatts - I don't work on this Integration but have forwarded on your comment to be looked into.

**Pull request merged**